doc/users/dflt_style_changes.rst

```diff
@@ -1065,6 +1065,17 @@ Z-order
 
 
 
+``ScalarFormatter`` tick label formatting with offsets
+======================================================
+
+With the default of ``rcParams['axes.formatter.useoffset'] = True``,
+an offset will be used when it will save 4 or more digits.  This can
```

Comment:
Example: an axis is showing years from 1990 through 2010.  This could be shown with ticks from -10 to 10 with an offset of 2000.   That is classic mode: 2 digits can be chopped off of the tick labels, so the offset is used if useoffset is True.  With the new setting, saving those two digits in the tick labels is not enough, so the offset won't be used, and users will be happier.  But if you have an axis from 20001 to 20003, then it will be given an offset of 20000, and the ticks will go from 1 to 3.  The tick labels have been shortened by 4 digits, so they will have a better chance of fitting in the frame without overlapping.

A version of this was put in place long ago.  Its purpose is to allow mpl to handle a wider range of cases by default, without excessively long tick labels.